lib/elixir/src/elixir_module.erl

```diff
@@ -292,14 +295,12 @@ lookup_attribute(DataSet, DataBag, Key) when is_atom(Key) ->
   end.
 
 warn_unused_attributes(File, DataSet, DataBag, PersistedAttrs) ->
-  Attrs = bag_lookup_element(DataBag, attributes, 2) -- [moduledoc | PersistedAttrs],
-
-  [case ets:lookup(DataSet, Key) of
-     [{_, _, Line}] when is_integer(Line) ->
-       elixir_errors:form_warn([{line, Line}], File, ?MODULE, {unused_attribute, Key});
-     _ ->
-       ok
-   end || Key <- lists:usort(Attrs)].
+  StoredAttrs = bag_lookup_element(DataBag, attributes, 2),
+  Attrs = [doc, typedoc, impl, since, deprecated | StoredAttrs -- PersistedAttrs],
+  Query = [{{Attr, '_', '$1'}, [{is_integer, '$1'}], [[Attr, '$1']]} || Attr <- lists:usort(Attrs)],
```

Comment:
Do we need to usort it now? I'm not sure if `ets:select` return duplicated documents if one document matches multiple times.

Reply:
I checked and indeed it doesn't return dupes.